data/store/mongo/mongo.go

```diff
@@ -913,3 +925,273 @@ func validateAndTranslateSelectors(selectors *data.Selectors) (bson.M, error) {
 
 	return selector, nil
 }
+
+// assumes all except freestyle is 5 minutes
+func getDuration(row *continuous.Continuous) int64 {
+    if strings.Contains(*row.DeviceID, "AbbottFreeStyleLibre") {
+        return 15
+    }
+    return 5
+}
+
+func (d *DataRepository) CalculateSummary(ctx context.Context, summary *data.Summary) (*data.Summary, error) {
+	if ctx == nil {
+		return nil, errors.New("context is missing")
+	}
+
+	if summary == nil {
+		return nil, errors.New("original summary is missing")
+	}
+
+	if summary.UserID == "" {
+        return nil, errors.New("summary missing UserID")
+    }
+
+	id := summary.UserID
+
+	var dataSet []*continuous.Continuous
+	selector := bson.M{
+		"_active": true,
+		"_userId": id,
+        "type": "cbg",
+	}
+
+    findOptions := options.Find()
+    findOptions.SetSort(bson.D{{"time", -1}})
+    findOptions.SetLimit(1)
+
+	cursor, err := d.Find(ctx, selector, findOptions)
+
+    if err != nil {
+		return nil, errors.Wrap(err, "unable to get last upload date")
+	}
+
+	if err = cursor.All(ctx, &dataSet); err != nil {
+		return nil, errors.Wrap(err, "unable to decode last upload date")
+	}
+
+    lastUpload, err := time.Parse(time.RFC3339Nano, *dataSet[0].CreatedTime)
```

Comment:
This looks very similar to the `GetLastUpdated` method below, with the exception of using different field for the `lastUpload`. Is that intentional? If the code should be same, then consider simply calling `GetLastUpdated`?

Reply:
Intentional, would need GetLastUpdated to just return the last record, and handle returning the Time value in the api code above it.
If it was modified to do that, it could then be generic enough for re-use, as we need both Time and CreatedTime fields for the summary data.

This seems like a worthy change for code reuse sake

Reply:
added both values to GetLastUpdated return, removed this repetition.